model-optimizer/mo/utils/cli_parser.py

```diff
@@ -792,6 +794,32 @@ def parse_input_value(input_value: str):
     return node_name, shape, value, data_type
 
 
+def get_layout_values(argv_layout: str):
+    node_name_re = r'[^(),\[\]]+'  # should fit any node name possible
+    source_layout_re = r'[^()>]+'  # should fit 'nhwc', '[n,h,w,c]' or 0312
+    target_layout_re = r'->[^()]+'  # should fit ->'layout'
+    # check if we deal with simple nameless case like 'nhwc' or more complicated like 'name(nhwc)'
+    p = re.compile('\S+\(\S+\)\S*')
+    if p.match(argv_layout):
+        p = re.compile(r'(({})\(({})({})?\),?)'.format(node_name_re, source_layout_re, target_layout_re))
+        m = p.findall(argv_layout)
+        res = dict()
+        validation_str = ''
+        for group in m:
+            group_3 = None
+            if group[3]:
+                # removing -> at the beggining of pattern
+                group_3 = group[3][2:]
+            validation_str += group[0]
+            res[group[1]] = {'source_layout': group[2], 'target_layout': group_3}
+        if validation_str != argv_layout:
+            raise Error("The layout can't be parsed")
+        return res
+    else:
```

Comment:
do we need custom parsing? I suppose we need to use `ov.Layout`

Reply:
ov.Layout is parsing the actual layout, this PR is about parsing command line argument.